### PR TITLE
Fix crio rpm installs

### PR DESCRIPTION
**Pull request by sdodson**

### CODE REVIEW

**Comment by sdodson** [comment]
Testing the new job here https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/8262/test_pull_request_openshift_ansible_extended_conformance_install_crio_rpm/1409/

File: sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml [hunk 1]
```diff
@@ -38,6 +38,35 @@ extensions:
...
+            major="$( echo "${branch}" | grep -Po "(?<=release-)[0-9](?=\.[0-9])" )"
+            minor="$( echo "${branch}" | grep -Po "(?<=release-[0-9]\.)[0-9]" )"
+            ;;
+          enterprise-* )
+            major="$( echo "${branch}" | grep -Po "(?<=release-)[0-9](?=\.[0-9])" )"
+            minor="$( echo "${branch}" | grep -Po "(?<=release-[0-9]\.)[0-9]" )"
+            ;;
+        esac
+        previous="$(( minor -1))"
+        sudo yum-config-manager --enable "rhel-7-server-ose-${major}.${minor}"
```
Comment (cevich) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml]: @sdodson you know more than I: Are all the above complications necessary?  Couldn't we simply rely on yum/rpm to always pick the latest version available.  In other words, just:  ``yum-config-manager --enable "rhel-7-server-*"`` or does that cause problems?

Comment (sdodson) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml]: We use the same jobs to install 3.9, 3.10, and future releases so we need to make sure we enable only the correct repos.  AFAIK cri-o 1.10 is not guaranteed to be compatible with Origin 3.9

Comment (cevich) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml]: OIC...thx for the explanation.  It seems like a lot of extra maintenance work, given the total number of jobs here (not just CRI-O).  Is this section at all common with other jobs?  (Just trying to think if I could help out writing some shared script or table or something)

**Comment by cevich** [comment]
So it seems the issue here is simply the lack of a 3.10 repository file.  Is there something I can do/look at to help move a solution for that forward?

**Comment by sdodson** [comment]
I think the commit I just pushed here and the change in https://github.com/openshift/origin-ci-tool/pull/170 should make this work

File: sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml [hunk 2]
```diff
@@ -38,6 +38,36 @@ extensions:
...
+        sudo yum-config-manager --disable origin-deps-rhel7\* || true
+        sudo yum-config-manager --disable rhel-7-server-ose\* || true
+
+        branch="${PULL_BASE_REF:-"master"}"
+        case "${branch}" in
+          master )
+            # 3.10
+            major=3
+            minor=10
+            ;;
```
Comment (cevich) [inline sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml]: This is the case I'm most worried about, maintenance wise.  It means when 3.11 comes around, some human needs to remember to edit this.  Is there no, like, ``VERSION`` file or something that could be used?  How about pulling it from ``git describe`` output?

Comment (cevich) [reply sjb/config/test_cases/test_branch_origin_extended_conformance_crio_rpm.yml]: idea:

```bash
    tag_version=$(git describe --abbrev | sed -r 's/^openshift-ansible-([0-9\.]+)-.+/\1/')
    major=$(echo $tag_version | cut -d . -f 1)
    minor=$(echo $tag_version | cut -d . -f 2)
```

**Comment by sdodson** [comment]
The AMI still lacks the repo definitions from https://github.com/openshift/origin-ci-tool/pull/170 checking on when we'll get a new AMI

**Comment by cevich** [comment]
Oh, I thought it would have happened by now, good to know.  Thx for following up Scott.

**Comment by sdodson** [comment]
Tagged a new image, this works now as evidenced by https://openshift-gce-devel.appspot.com/build/origin-ci-test/pr-logs/pull/openshift_openshift-ansible/8618/test_pull_request_openshift_ansible_extended_conformance_install_crio_rpm/2359/

**Pull request merged**